index.bs

```diff
@@ -1444,19 +1441,46 @@ Because of that, JavaScript authors take for granted
 that the data available to a function won’t change unexpectedly
 while the function is running.
 
-So if a JavaScript Web API exposes some piece of data,
-such as an object property,
-the user agent must not update that data
-while a JavaScript task is running.
-Instead, if the underlying data changes,
-<a>queue a task</a> to modify the exposed version of the data.
+Changes that are not the result of developer action
+and changes that are asynchronously delivered
+should not happen in the middle of other JavaScript,
+including between [=microtasks=].
+Instead, a task should be [[html#queuing-tasks|queued]],
+or they should update as part of [=update the rendering=].
+
+<div class="example">
+During a `while (true)`,
+and after `await`ing an already-resolved `Promise`,
+you *shouldn't* expect things like:
+
+* The DOM to update as a result of the HTML parser loading network content
+* {{HTMLImageElement/width|img.width}} to change as a result of loading image data from the network
+* Buttons of a {{Gamepad}} to change state
+* {{Element/scrollTop}} to change, even if scrolling can visually occur
+
+These things aren't updated by the currently running script,
+so they shouldn't change during the current task.
+
+</div>
+
+Data can update synchronously from the result of developer action.
 
 <div class="example">
-If a JavaScript task has accessed the {{NavigatorOnline/onLine|navigator.onLine}} property,
-and browser's online status changes,
-the property won't be updated until the next task runs.
+{{ChildNode/remove()|node.remove()}} changes the DOM synchronously
+and is immediately observable.
 </div>
 
+A few kinds of situations justify violating this rule:
+
+* Observing the current time,
+    as in {{Date/now|Date.now()}} and {{Performance/now|performance.now()}},
+    although note that it's also useful to present a consistent task-wide time
+    as in {{AnimationTimeline/currentTime|document.timeline.currentTime}}.
+* Functions meant to help developers interrupt synchronous work,
+    as in the case of the proposed {{isInputPending()}}.
+* States meant to protect users from surprising UI changes,
+    like [=transient activation=].
```

Comment:
I don't think you can get in a state there where `x.isActive !== x.isActive`?

Reply:
If I'm reading it correctly, Chromium's `isActive` bottoms out at https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/common/frame/user_activation_state.cc;l=81;drc=616d60fca655937c2b730db94fd32d37ddff3bb5

```c++
bool UserActivationState::IsActiveInternal() const {
  return base::TimeTicks::Now() <= transient_state_expiry_time_;
}
```

I didn't notice any caching, which means that can return false immediately after it returned true, within the same line of JS code. @mustaqahmed, can you confirm whether I'm reading that correctly? 

This intra-task change also seems to be the semantics that @jan-ivar is saying should be acceptable in https://github.com/w3ctag/design-principles/pull/496/files#r1820794873.

FWIW, putting users ahead of authors in this way seems to be correct to me.

Reply:
I guess you're right. It seems really bad that this is a getter then and not a method.

Reply:
I've pointed that out in fac9f74fecf71a3469fc05ba6951111ab4f22be6, although it's possible that's too much emphasis.